Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -48,17 +48,21 @@ class BTAnalyticsService: Equatable {
     ///  Events are queued and sent in batches to the analytics service, based on the status of the app and
     ///  the number of queued events. After exiting this method, there is no guarantee that the event has been sent.
     /// - Parameter eventName: String representing the event name
-    func sendAnalyticsEvent(_ eventName: String) {
+    func sendAnalyticsEvent(_ eventName: String, errorDescription: String? = nil) {
```

Comment:
We can look at this later, but this method looks like it is only called within the tests. This is a little odd to me since that means we're basically never checking for the flush threshold before sending an API request with events.

Reply:
Yeah, I just jumped into v5 to poke around at this and we were doing the exact same thing there. We have the method:
```
- (void)sendAnalyticsEvent:(NSString *)eventKind {
    dispatch_async(dispatch_get_main_queue(), ^{
        [self enqueueEvent:eventKind];
        [self checkFlushThreshold];
    });
}
```
in Obj-C and the only callers are our tests. We should certainly look into this and remove it (if possible) or move the logic we want into the other flush method. We can 🏈 for now but agree we should revisit.

Reply:
Ah, actually it looks like the caller was these events that used to exist in APIClient `[self queueAnalyticsEvent:@"ios.started.client-key"];`. Which I guess makes sense for how that event was tied to everything. Now that we removed the events in APIClient, we can likely clean these methods up entirely.